alert-common/src/main/java/com/synopsys/integration/alert/common/channel/issuetracker/service/IssueTrackerRequestCreator.java

```diff
@@ -0,0 +1,157 @@
+            if (ItemOperation.ADD == operation || ItemOperation.UPDATE == operation) {
+                IssueContentModel issueContentModel = issueTrackerMessageParser.createIssueContentModel(providerContentKey.getProviderName(), IssueCreationRequest.OPERATION, topic, nullableSubTopic, componentItems, arbitraryItem);
+                issueRequest = IssueCreationRequest.of(issueSearchProperties, issueContentModel, alertIssueOrigin);
+            } else if (ItemOperation.DELETE == operation) {
+                IssueContentModel issueContentModel = issueTrackerMessageParser.createIssueContentModel(providerContentKey.getProviderName(), IssueResolutionRequest.OPERATION, topic, nullableSubTopic, componentItems, arbitraryItem);
+                issueRequest = IssueResolutionRequest.of(issueSearchProperties, issueContentModel, alertIssueOrigin);
+            } else if (ItemOperation.INFO == operation) {
+                IssueContentModel issueContentModel = issueTrackerMessageParser.createIssueContentModel(providerContentKey.getProviderName(), IssueCommentRequest.OPERATION, topic, nullableSubTopic, componentItems, arbitraryItem);
+                issueRequest = IssueCommentRequest.of(issueSearchProperties, issueContentModel, alertIssueOrigin);
+            }
+            if (null != issueRequest) {
+                issues.add(issueRequest);
+            }
+        }
+        return issues;
+    }
+
+    protected String createAdditionalTrackingKey(ComponentItem componentItem) {
+        if (null != componentItem && !componentItem.collapseOnCategory()) {
+            LinkableItem categoryItem = componentItem.getCategoryItem();
+            return categoryItem.getName() + categoryItem.getValue();
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private List<IssueTrackerRequest> createRequestsPerComponent(IssueConfig issueConfig, ProviderMessageContent messageContent) throws IntegrationException {
+        LinkableItem topic = messageContent.getTopic();
+        LinkableItem nullableSubTopic = messageContent.getSubTopic().orElse(null);
+
+        ContentKey providerContentKey = messageContent.getContentKey();
+        String providerUrl = messageContent.getProvider().getUrl()
+                                 .map(this::formatProviderUrl)
+                                 .orElse("");
+        List<IssueTrackerRequest> requests;
+        if (messageContent.isTopLevelActionOnly()) {
+            requests = createUpdateRequestsByTopLevelAction(issueConfig, providerContentKey, providerUrl, topic, nullableSubTopic, messageContent.getAction().orElse(ItemOperation.INFO));
+        } else {
+            requests = createRequestsByComponentGroup(providerContentKey, providerUrl, topic, nullableSubTopic, messageContent.groupRelatedComponentItems());
+        }
+        return requests;
+    }
+
+    private String formatProviderUrl(String originalUrl) {
```

Comment:
Seems like this would be better suited in a Utility class of some kind

Reply:
It was in a jira-specific utility class before, but it didn't belong there. This class is going to get some more brushing up in a future PR, but for now, I'm trying to avoid scope creep on things not directly related to completing Azure functionality.